File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    dic = {}
    for str in keymap:
        chars = list(str)
        for i in range(len(chars)):
            if chars[i] not in dic.keys():
                dic[chars[i]] = i+1
            elif i+1 < dic[chars[i]]:
                dic[chars[i]] = i+1
                
    answer = [0] * len(targets)
    
    for i in range(len(targets)):
        for ch in list(targets[i]):
            if ch in dic.keys():
                answer[i] += dic[ch]
            else:
                answer[i] = -1
                break   
                
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    
    var keys = {};
    
    for(key of keymap) {
        for (var idx = 0; idx < key.length; idx++) {
            if (key[idx] in keys && idx+1 < keys[key[idx]]) {
                keys[key[idx]] = idx+1;
            }else if (!(key[idx] in keys)) {
                keys[key[idx]] = idx+1;
            }
        }
    }
    console.log(keys);
    
    
    
    
    var answer = [];
    for (var target of targets) {
        var count = 0;
        for(var ca of [...target]) {
            if (ca in keys) {
                count += keys[ca];
            }else {
                count = -1;
                break;
            }
        }
        answer.push(count);
    }
    return answer;
}